packages/ses/src/lockdown.js

```diff
@@ -242,7 +272,35 @@ export const repairIntrinsics = (options = {}) => {
   // trace retained:
   priorRepairIntrinsics.stack;
 
-  assertDirectEvalAvailable();
+  const { functionAllowed, evalAllowed, directEvalAllowed } =
+    probeHostEvaluators();
+
+  // This could be a strict Content Security Policy containing either a `default-src` or a `script-src` directive, or an ES host with broken APIs.
+  const noEvaluators = !evalAllowed && !functionAllowed; // eval() itself and the Function() constructor are not allowed to execute.
+
+  hostEvaluators === 'all' &&
+    noEvaluators &&
```

Comment:
more precisely, if we want to cover a situation where just one of the evaluators we check for is broken:
```suggestion
    !(evalAllowed && functionAllowed) &&
```